bin/eland_import_hub_model

```diff
@@ -93,36 +112,100 @@ def main():
+            es_args['api_key'] = cli_args.es_api_key
+        elif cli_args.es_username:
+            if not cli_args.es_password:
+                logging.error(f"Password for user {cli_args.es_username} was not specified.")
+                exit(1)
+
+            es_args['basic_auth'] = (cli_args.es_username, cli_args.es_password)
 
-    es = elasticsearch.Elasticsearch(args.url, request_timeout=300, verify_certs=args.insecure, ca_certs=args.ca_certs)  # 5 minute timeout
+        es_client = Elasticsearch(args.url, **es_args)
+        es_info = es_client.info()
+        logger.info(f"Connected to cluster named '{es_info['cluster_name']}' (version: {es_info['version']['number']})")
 
-    # trace and save model, then upload it from temp file
+        return es_client
+    except AuthenticationException as e:
+        logger.error(e)
+        exit(1)
+
+
+if __name__ == "__main__":
+    # Configure logging
+    logging.basicConfig(format='%(asctime)s %(levelname)s : %(message)s')
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+
+    try:
+        from eland.ml.pytorch import PyTorchModel
+        from eland.ml.pytorch.transformers import SUPPORTED_TASK_TYPES, TransformerModel
+    except ModuleNotFoundError as e:
+        logger.error(textwrap.dedent(f"""\
+            \033[31mFailed to run because module '{e.name}' is not available.\033[0m
+            
+            This script requires PyTorch extras to run. You can install these by running:
+            
+                \033[1m{sys.executable} -m pip install 'eland[pytorch]'
+            \033[0m"""))
+        exit(1)
+
+    # Parse arguments
+    args = get_arg_parser().parse_args()
+
+    # Connect to ES
+    logger.info("Establishing connection to Elasticsearch")
+    es = get_es_client(args)
+
+    # Trace and save model, then upload it from temp file
     with tempfile.TemporaryDirectory() as tmp_dir:
-        print(f"Loading HuggingFace transformer tokenizer and model {args.hub_model_id}")
+        logger.info(f"Loading HuggingFace transformer tokenizer and model '{args.hub_model_id}'")
+
         tm = TransformerModel(args.hub_model_id, args.task_type, args.quantize)
         model_path, config_path, vocab_path = tm.save(tmp_dir)
 
-        es_model_id = (
-            args.elasticsearch_model_id
-            if args.elasticsearch_model_id
-            else tm.elasticsearch_model_id()
-        )
-
-        ptm = PyTorchModel(es, es_model_id)
-        if args.clear_previous:
-            print(f"Stopping previous deployment and deleting model: {ptm.model_id}")
-            ptm.stop()
-            ptm.delete()
-        print(f"Importing model: {ptm.model_id}")
-        ptm.import_model(model_path, config_path, vocab_path)
-
-    # start the deployed model
+        ptm = PyTorchModel(es, args.es_model_id if args.es_model_id else tm.elasticsearch_model_id())
+        model_exists = es.ml.get_trained_models(model_id=ptm.model_id, ignore=404).meta.status == 200
```

Comment:
Should use `es.options(ignore_status=404).ml.get_trained_models(...)` instead of `ignore=404` which is deprecated.